src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/legacy/Example2.java

```diff
@@ -0,0 +1,14 @@
+/*xml
+<module name="Checker">
+  <module name="JavadocPackage">
+    <property name="allowLegacy" value="true"/>
+  </module>
+</module>
+*/
+// xdoc section -- start
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocpackage.legacy;
+// OK as package.html file is present in directory
```

Comment:
I hope you see yourself that generated documentation is weird as it reference to something that use does not see.
We need to print content of folder:
https://github.com/checkstyle/checkstyle/tree/a09d8d8cfa81b6d5a3115fd75ed85c866ac04514/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocpackage/legacy

lets put in template file one more code block that show structure of folder:
```
Example2.java
package.html
``` 

please do for all examples.
Content of java files does not really matter, it is only useful to show that violation is placed on java files, even something is missed in folder.

So each example will he have 3 code blocks:
1) config macros
2) folder structure (hardcoded in as plain tags in xdoc template file)
3) java file content macros

Reply:
Done